src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -186,13 +183,97 @@ private void applyFix(final String polluterName,
         //        }
     }
 
+    // The delta debugging logic to return minimal list of statements to add in to make victim pass
+    // In addition to polluter and victim to run, need also list of statements; assumed statements already make victim pass
+    // The value n is needed to represent granularity of partitioning the statements to get minimal list
+    private NodeList<Statement> deltaDebug(final String polluterName,
```

Comment:
Not really clear to me what `n` is exactly. For example, why do we pass in a value of `2` in the initial call in applyFix? Why not `1`? And there are recursive calls with various values of `n` (`n - 1`, `2` again, `n * 2`). How are we sure this will be a minimal list of statements?

Reply:
The n is a part of the delta-debugging logic to find the minimal set of elements. The n represents the number of chunks to try at a time to split the elements into for minimizing, so starting with n=2 makes it start with binary search. The granularity increases when things fail, forcing it to split into more chunks (so fewer elements in each chunk) to see if it can find another chunk that works. When there are fewer statement remaining than there are acceptable number of chunks, then we've reached the end.